sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionEvaluatesToConstant.cs

```diff
@@ -194,20 +194,15 @@ private static SyntaxNode GetUnreachableStatement(IEnumerable<ExpressionSyntax>
                     : node.Parent) // Constant node is the only expression in an if statement condition
                 .FirstOrDefault();
 
-            if (constantValue)
-            {
-                return (SyntaxNode)
+            return constantValue
+                ? (SyntaxNode)
                     (parent as ConditionalExpressionSyntax)?.WhenFalse ??
-                    (parent as IfStatementSyntax)?.Else?.Statement;
-            }
-            else
-            {
-                return (SyntaxNode)
+                    (parent as IfStatementSyntax)?.Else?.Statement
+                : (SyntaxNode)

```

Comment:
I kept it because there was quite a lot in common but happy to undo if you prefer.